src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Cmdlets/GetAzureRmVMGuestPolicyReport.cs

```diff
@@ -0,0 +1,180 @@
+        public string InitiativeName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.InitiativeId)]
+        [ValidateNotNullOrEmpty]
+        public string InitiativeId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ReportIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.ReportId)]
+        [ValidateNotNullOrEmpty]
+        public string ReportId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        public SwitchParameter Detailed { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = ParameterHelpMessages.LatestSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3,  HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        public SwitchParameter Latest { get; set; }
+
+        /// <summary>
+        /// Executes the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            IEnumerable<GuestConfigurationPolicyAssignment> gcPolicyAssignments = null;
+            IEnumerable<GuestConfigurationPolicyAssignmentReport> gcPolicyAssignmentReports = null;
+
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.InitiativeNameScope:
+                    if (Detailed.IsPresent)
+                    {
+                        gcPolicyAssignmentReports = GetAllGuestConfigurationAssignmentReportsByInitiativeName(ResourceGroupName, VMName, InitiativeName, Latest.IsPresent);
+                        if(gcPolicyAssignmentReports == null || gcPolicyAssignmentReports.Count() == 0)
+                        {
+                            var message = string.Format(UserMessages.NoDetailedGuestConfigurationPolicyReportsFoundInitiativeName, ResourceGroupName, VMName, InitiativeName);
+                            WriteObject(message);
```

Comment:
In this case, you should return nothing.

Reply:
If you like, you can write a message to the verbose stream, but you should not write a string here.